### PR TITLE
Release 9.12.1-alpha2

### PR DESCRIPTION
**Pull request by wz1000**

### CODE REVIEW

**Comment by hasufell** [comment]
https://github.com/haskell/ghcup-metadata/actions/runs/11684445548

File: ghcup-prereleases-0.0.8.yaml [hunk 1]
```diff
@@ -2141,13 +2141,163 @@ ghcupDownloads:
...
+              dlSubdir: ghc-9.12.0.20241031-x86_64-unknown-linux
+              dlUri: https://downloads.haskell.org/~ghc/9.12.0.20241031/ghc-9.12.0.20241031-x86_64-ubuntu18_04-linux.tar.xz
+            '>= 20': &ghc9121alpha2-x86_64-ubuntu22_04
+              dlHash: a0dc0c19509a738acfe3b75d284bda1b2e4a07e43b36a46f9f305985372028eb
+              dlOutput: ghc-9.12.0.20241031-x86_64-linux-ubuntu20_04.tar.xz
+              dlSubdir: ghc-9.12.0.20241031-x86_64-unknown-linux
+              dlUri: https://downloads.haskell.org/~ghc/9.12.0.20241031/ghc-9.12.0.20241031-x86_64-ubuntu20_04-linux.tar.xz
+            unknown_versioning: *ghc9121alpha2-x86_64-ubuntu22_04
+          Linux_RedHat:
+            unknown_versioning: *ghc9121alpha2-x86_64-centos7
```
Comment (hasufell) [inline ghcup-prereleases-0.0.8.yaml:2238]: It fails for RedHat 9, which apparently doesn't have libtinfo.so.5 anymore: https://github.com/haskell/ghcup-metadata/actions/runs/11684445548/job/32535716297#step:5:15963

Comment (hasufell) [reply ghcup-prereleases-0.0.8.yaml:2238]: You can check here about binary compatibility: https://gitlab.haskell.org/ghc/ghc/-/wikis/platforms/linux

It appears that "Ubuntu 20.04 (focal)" might work.

You can test it in the docker image `registry.access.redhat.com/ubi9/ubi` locally.

Comment (hasufell) [reply ghcup-prereleases-0.0.8.yaml:2238]: Additionally... GHC HQ could consider building Rocky9 bindists (which are binary compatible with RedHat9).

Comment (hasufell) [reply ghcup-prereleases-0.0.8.yaml:2238]: it looks like ghc-9.12.0.20241031-x86_64-fedora33-linux.tar.xz might actually work better

Comment (wz1000) [reply ghcup-prereleases-0.0.8.yaml]: I've changed it to use fedora33 for redhat.

File: ghcup-prereleases-0.0.8.yaml [hunk 2]
```diff
@@ -2235,7 +2235,7 @@ ghcupDownloads:
               dlUri: https://downloads.haskell.org/~ghc/9.12.0.20241031/ghc-9.12.0.20241031-x86_64-ubuntu20_04-linux.tar.xz
             unknown_versioning: *ghc9121alpha2-x86_64-ubuntu22_04
           Linux_RedHat:
-            unknown_versioning: *ghc9121alpha2-x86_64-centos7
+            unknown_versioning: *ghc9121alpha2-x86_64-fedora33
```
Comment (hasufell) [inline ghcup-prereleases-0.0.8.yaml:2238]: What I meant is:

* centos7 for RedHat 8
* fedora33 for RedHat 9 (and unknown/later)

Comment (wz1000) [reply ghcup-prereleases-0.0.8.yaml]: done.

File: ghcup-prereleases-0.0.8.yaml [hunk 3]
```diff
@@ -2141,13 +2141,165 @@ ghcupDownloads:
         dlSubdir: ghc-9.12.20241014
         dlUri: https://downloads.haskell.org/~ghc/9.12.20241014/ghc-9.12.20241014-src.tar.xz
       viTags:
-      - LatestPrerelease
```
Comment (hasufell) [inline ghcup-prereleases-0.0.8.yaml:2144]: When removing `LatestPrerelease` you have to add `Prerelease` tag.

Comment (wz1000) [reply ghcup-prereleases-0.0.8.yaml:2144]: done

**Pull request merged**